### PR TITLE
22w43a Added Partial Math Mappings

### PR DESCRIPTION
**Pull request by kevinthegreat1**
Someone actually good at linear algebruh will have to finish MatrixUtil.
But there are some mappings in here.

### CODE REVIEW

File: mappings/net/minecraft/util/math/Division.mapping [hunk 1]
```diff
@@ -0,0 +1,16 @@
+CLASS net/minecraft/class_7836 net/minecraft/util/math/Division
+	COMMENT Calculates {@link #divisor} number of integers that are floor or ceiling of the result of the division and average to the exact result of the division.
+	COMMENT The first returned integer always equals to {@code Math.floor(divident / divisor)}
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/Division.mapping]: Your explanation is a bit too focusing on implementation imo. How about this?

```suggestion
	COMMENT A class for dividing an integer into {@link #divisor} numbers of
	COMMENT integers such that the difference between any integers is less than or
	COMMENT equal to {@code 1}. For example, {@code new Division(13, 5)} will
	COMMENT produce {@code 2, 3, 2, 3, 3}.
```

File: mappings/net/minecraft/util/math/Division.mapping [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	FIELD field_40738 divisor I
+	FIELD field_40739 quotient I
+	FIELD field_40740 mod I
+	FIELD field_40741 returnedCount I
+	FIELD field_40742 remainder I
+	METHOD <init> (II)V
+		ARG 1 dividend
+		ARG 2 divisor
+	METHOD hasNext hasNext ()Z
+	METHOD method_46402 asInterable (II)Ljava/lang/Iterable;
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/Division.mapping]: Typo? unless you mean ***int***erable is iterable of integer.

```suggestion
	METHOD method_46402 asIterable (II)Ljava/lang/Iterable;
```

File: mappings/net/minecraft/util/math/MatrixUtil.mapping [hunk 3]
```diff
@@ -0,0 +1,9 @@
+CLASS net/minecraft/class_7837 net/minecraft/util/math/MatrixUtil
+	FIELD field_40747 COS_PI_OVER_8 F
+	FIELD field_40748 SIN_PI_OVER_8 F
+	METHOD method_46412 svdDecompose (Lorg/joml/Matrix3f;)Lorg/apache/commons/lang3/tuple/Triple;
+	METHOD method_46413 affineTransform (Lorg/joml/Matrix4f;)Lorg/joml/Matrix4x3f;
+		ARG 0 matrix4f
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/MatrixUtil.mapping]:
```suggestion
		ARG 0 matrix
```

File: mappings/net/minecraft/util/math/MatrixUtil.mapping [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+	METHOD method_46414 scale (Lorg/joml/Matrix4f;F)Lorg/joml/Matrix4f;
+		ARG 0 matrix4f
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/MatrixUtil.mapping]:
```suggestion
		ARG 0 matrix
```

File: mappings/net/minecraft/util/math/Division.mapping [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	COMMENT The first returned integer always equals to {@code Math.floor(divident / divisor)}
+	FIELD field_40738 divisor I
+	FIELD field_40739 quotient I
+	FIELD field_40740 mod I
+	FIELD field_40741 returnedCount I
+	FIELD field_40742 remainder I
+	METHOD <init> (II)V
+		ARG 1 dividend
+		ARG 2 divisor
+	METHOD hasNext hasNext ()Z
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/Division.mapping]: We don't map non-obfuscated methods.

```suggestion
```

File: mappings/net/minecraft/util/math/Division.mapping [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+	METHOD method_46402 asInterable (II)Ljava/lang/Iterable;
+		ARG 0 dividend
+		ARG 1 divisor
+	METHOD nextInt nextInt ()I
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/Division.mapping]:
```suggestion
```

File: mappings/net/minecraft/util/math/RotationAxis.mapping [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+	METHOD method_46353 (F)Lorg/joml/Quaternionf;
+		ARG 0 rad
+	METHOD method_46354 (F)Lorg/joml/Quaternionf;
+		ARG 0 rad
+	METHOD method_46355 (F)Lorg/joml/Quaternionf;
+		ARG 0 rad
+	METHOD method_46356 of (Lorg/joml/Vector3f;)Lnet/minecraft/class_7833;
+		ARG 0 axis
+	METHOD rotation rotation (F)Lorg/joml/Quaternionf;
+		ARG 1 rad
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/RotationAxis.mapping]:
```suggestion
	METHOD rotation (F)Lorg/joml/Quaternionf;
		ARG 1 rad
```

File: mappings/net/minecraft/util/math/RotationAxis.mapping [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+	METHOD rotationDegrees rotationDegrees (F)Lorg/joml/Quaternionf;
+		ARG 1 deg
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/RotationAxis.mapping]:
```suggestion
	METHOD rotationDegrees (F)Lorg/joml/Quaternionf;
		ARG 1 deg
```

**Review by enbrain — CHANGES_REQUESTED** [review]

**Comment by kevinthegreat1** [comment]
I updated the Javadoc to something similar. I also changed the name to divider as it literally divides an integer.

**Review by enbrain — APPROVED** [review]

File: mappings/net/minecraft/util/math/Divider.mapping [hunk 9]
```diff
@@ -0,0 +1,16 @@
+CLASS net/minecraft/class_7836 net/minecraft/util/math/Divider
+	COMMENT A class for dividing an integer into {@link #divisor} number of integers
+	COMMENT such that the difference between any integers is {@code 0} or {@code 1}.
+	COMMENT The resulting integers sum to the {@code dividend}.
+	COMMENT For example, {@code new Division(13, 5)} will produce {@code 2, 3, 2, 3, 3}.
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/Divider.mapping]:
```suggestion
	COMMENT For example, {@code new Divider(13, 5)} will produce {@code 2, 3, 2, 3, 3}.
```

**Review by apple502j — APPROVED** [review]

**Pull request merged**